src/completion_matcher.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <string_view.h>
+
+class CompletionMatcher {
+ public:
+  // virtual ~CompletionMatcher() = default; // don't know why but it crashed if I uncomment this like, investigating
+  virtual int Match(std::string_view text) = 0;
```

Comment:
Can you add some timing to see if the virtual function call is negatively impacting how long it takes to filter completion results?

Reply:
I'll check it a bit later, no problem. I'm sure that virtual function call will not spend much more time than plain function call, especially comparing to all that fuzzy-matching stuff = )